### ISSUE TITLE
Consider having one unified interface for the filters (SG/NACL) and other connection rules resources (transit gateway)

### ISSUE THREAD

**Issue by ShiriMoran** [np-guard/vpc-network-config-analyzer#504]
Consider re-organizing functionality related merely to explian tgw, and merging similar structs for rules/prefixes indexes
The function `StringPrefixDetails` is more explainability-specific...  the relevant part from TGW to return is the prefilx filter object and its indexes. 
maybe should consider to separate those functionalities (also for filter traffic resources), consider a separate issue.. 
(also the prefix filter index is some other "rule" similar to rule index already captured for nacl/sg rules here... )
maybe the abstract type `RulesInFilter` should be modified to capture also rules for tgw objects? 

_Originally posted by @adisos in https://github.com/np-guard/vpc-network-config-analyzer/pull/492#discussion_r1555352574_

**Comment by ShiriMoran** [comment]
>  (also the prefix filter index is some other "rule" similar to rule index already captured for nacl/sg rules here... ) maybe the abstract type `RulesInFilter` should be modified to capture also rules for tgw objects?
> 
The tgw object equiv of `RulesInFilter` is not stored at the explainability struct. Only the relevant tgw router is stored at the struct; the deduction of the prefix filter here is more straightforward - no multiple tables and layers as in the filter case. Having said that, it might make sense to save the relevant details in the struct - in which case using `RulesInFilter` could indeed make sense. It will require listing the tgw connection in the same column as SG and NACL. Not completely sure, perhaps we should discuss.

**Comment by ShiriMoran** [comment]
> The function `StringPrefixDetails` is more explainability-specific... the relevant part from TGW to return is the prefilx filter object and its indexes. maybe should consider to separate those functionalities (also for filter traffic resources), consider a separate issue.. 

1. Its not only the indexes, its also the `lt` `gt` properties and the `action`; it is internal to the transit gateway and I'm not sure we want to expose these specific details outside of the IBMVPC package
2. In the filters case we use an internal (to SG/NACL) functionality to print - specifically, `getSGRule` and `getNACLRule`. This functionality is not only for explainability and we can use it, of course, only from within the internal SG and NACL functionality.

**Comment by ShiriMoran** [comment]
after discussing with @zivnevo and @adisos it was agreed to add an interface for filters that will be implemented by `NACLLayer`, `SecurityGroupLayer` and `TransitGateway`; functionality used by Explainability should be moved to this interface. 
Once this Interface is in place, it should make sense to add  `crossVPC rulesInLayers`  to `rulesConnection`

**Comment by ShiriMoran** [comment]
solved partly by #574 
We will revisit once there are more rule based resources
For now lowering priority
@adisos

**Comment by ShiriMoran** [comment]
Consider doing it after https://github.com/np-guard/vpc-network-config-analyzer/pull/675 is merge, by defining an interface of `[]RuleOfFilter` (define type) that will implement functionality to replace `StringDetailsOfRules(...)` and `ListFilterWithAction(...)`
